tests/test_validation.py

```diff
@@ -7,14 +7,17 @@
 import numpy as np
 
 
-@pytest.fixture(scope='module')
-def source_1bin_example1():
+def get_source_1bin_example1():
     with open('validation/data/1bin_example1.json') as read_json:
         return json.load(read_json)
 
 
 @pytest.fixture(scope='module')
-def spec_1bin_shapesys(source=source_1bin_example1()):
+def source_1bin_example1():
```

Comment:
@kratsg I think I'm missing something obvious. This fixture doesn't ever show up again

```
$ git grep "source_1bin_example1"
tests/test_validation.py:def get_source_1bin_example1():
tests/test_validation.py:def source_1bin_example1():
tests/test_validation.py:    return get_source_1bin_example1()
tests/test_validation.py:def get_spec_1bin_shapesys(source=get_source_1bin_example1()):
tests/test_validation.py:    source=get_source_1bin_example1(),
tests/test_validation.py:    spec=get_spec_1bin_shapesys(get_source_1bin_example1()),
tests/test_validation.py:    source=get_source_1bin_example1(),
```

so how does this work? I'll go re-read on how module level fixtures work but if you can briefly summarize that helps too.

Reply:
Yeah, after reading the "[Scope: sharing fixtures across classes, modules, packages or session](https://docs.pytest.org/en/stable/fixture.html#scope-sharing-fixtures-across-classes-modules-packages-or-session)" section of the pytest docs I'm still confused as this fixture is never used.

Reply:
I'm just moving it to the appropriate fixture pattern that's suggested. This first move will eventually indicate that certain fixtures become unused (which means they never should've been fixtures in the first place).

Reply:
Why leave them in then and not just remove them in this PR? I'm a bit confused by the renaming to `source_X` to `get_source_X` everywhere

```diff
@@ -7,14 +7,17 @@ from pathlib import Path
 import numpy as np
 
 
-@pytest.fixture(scope='module')
-def source_1bin_example1():
+def get_source_1bin_example1():
     with open('validation/data/1bin_example1.json') as read_json:
         return json.load(read_json)
 
 
 @pytest.fixture(scope='module')
-def spec_1bin_shapesys(source=source_1bin_example1()):
+def source_1bin_example1():
+    return get_source_1bin_example1()
+
+
+def get_spec_1bin_shapesys(source=get_source_1bin_example1()):
     spec = {
         'channels': [
             {
@@ -46,7 +49,11 @@ def spec_1bin_shapesys(source=source_1bin_example1()):
```

Why all the renaming if it seems the only functional change is what used to be called `source_1bin_example1` is no longer a fixture?

> I'm just moving it to the appropriate fixture pattern that's suggested.

Can you link to what the suggestion is, as maybe that will explain my questions?

Reply:
https://docs.pytest.org/en/stable/deprecations.html

I’m not going to rewrite the validation tests in this PR. I’m simply updating them with the minimal amount of work that follows pytest docs and works.

Reply:
Okay I'll have to look at this later in more depth then. I don't understand why it is necessary to add all these `get_X` functions and keep `source_X` fixtures that are never used when the following patch would allow `setup_1bin_shapesys` to work as the fixture you need while removing a lot of code.

I suspect I'm just misunderstanding what it is that you're trying to do, so can you give me an example of why this is necessary (for a future PR or something).

```diff
diff --git a/tests/test_validation.py b/tests/test_validation.py
index ed1eb9b8..a2393173 100644
--- a/tests/test_validation.py
+++ b/tests/test_validation.py
@@ -7,17 +7,12 @@ from pathlib import Path
 import numpy as np
 
 
-def get_source_1bin_example1():
+def source_1bin_example1():
     with open('validation/data/1bin_example1.json') as read_json:
         return json.load(read_json)
 
 
-@pytest.fixture(scope='module')
-def source_1bin_example1():
-    return get_source_1bin_example1()
-
-
-def get_spec_1bin_shapesys(source=get_source_1bin_example1()):
+def spec_1bin_shapesys(source=source_1bin_example1()):
     spec = {
         'channels': [
             {
@@ -48,12 +43,7 @@ def get_spec_1bin_shapesys(source=get_source_1bin_example1()):
     return spec
 
 
-@pytest.fixture(scope='module')
-def spec_1bin_shapesys():
-    return get_spec_1bin_shapesys()
-
-
-def get_expected_result_1bin_shapesys(mu=1.0):
+def expected_result_1bin_shapesys(mu=1.0):
     if mu == 1:
         expected_result = {
             "exp": [
@@ -68,28 +58,22 @@ def get_expected_result_1bin_shapesys(mu=1.0):
     return expected_result
 
 
-@pytest.fixture(scope='module')
-def expected_result_1bin_shapesys():
-    return get_expected_result_1bin_shapesys()
-
-
-def get_setup_1bin_shapesys(
-    source=get_source_1bin_example1(),
-    spec=get_spec_1bin_shapesys(get_source_1bin_example1()),
-    mu=1,
-    expected_result=get_expected_result_1bin_shapesys(1.0),
-    config={'init_pars': 2, 'par_bounds': 2},
-):
-    return {
-        'source': source,
-        'spec': spec,
-        'mu': mu,
-        'expected': {'result': expected_result, 'config': config},
-    }
-
-
 @pytest.fixture(scope='module')
 def setup_1bin_shapesys():
+    def get_setup_1bin_shapesys(
+        source=source_1bin_example1(),
+        spec=spec_1bin_shapesys(source_1bin_example1()),
+        mu=1,
+        expected_result=expected_result_1bin_shapesys(1.0),
+        config={'init_pars': 2, 'par_bounds': 2},
+    ):
+        return {
+            'source': source,
+            'spec': spec,
+            'mu': mu,
+            'expected': {'result': expected_result, 'config': config},
+        }
+
     return get_setup_1bin_shapesys()
 
 
@@ -158,7 +142,7 @@ def expected_result_1bin_lumi():
 
 
 def get_setup_1bin_lumi(
-    source=get_source_1bin_example1(),
+    source=source_1bin_example1(),
     spec=get_spec_1bin_lumi(),
     mu=1,
     expected_result=get_expected_result_1bin_lumi(1.0),
```